### PR TITLE
[sample](doris-soruce-demo)add read doris  demo by thrift 

### PR DESCRIPTION
**Pull request by JNSimba**
## Proposed changes
issue：[#2](https://github.com/apache/doris/issues/20136)
pick pr： https://github.com/apache/doris/pull/20123

add read doris  demo by thrift 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: samples/doris-demo/doris-source-demo/src/main/java/org/apache/doris/sdk/DorisReaderExample.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    public static void main(String[] args) throws Exception {
+        JSONObject queryPlan = getQueryPlan();
+        readData(queryPlan);
+        System.out.println(result);
+        System.out.println(result.size());
+    }
+
+    private static JSONObject getQueryPlan() throws Exception {
+        try (CloseableHttpClient client = HttpClients.custom().build()) {
+            HttpPost put = new HttpPost(String.format("http://%s/api/%s/%s/_query_plan", dorisUrl, database, table));
```
Comment (xiaokang) [inline samples/doris-demo/doris-source-demo/src/main/java/org/apache/doris/sdk/DorisReaderExample.java:73]: suggestion name: post

File: samples/doris-demo/doris-source-demo/src/main/java/org/apache/doris/sdk/DorisReaderExample.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+            client.closeScanner(closeParams);
+            if ((transport != null) && transport.isOpen()) {
+                transport.close();
+            }
+            //System.out.println(String.format("read tablet %s from backend %s finished", tabletId, routingsBackend));
+        }
+    }
+
+    private static int convertArrow(TScanBatchResult nextResult, List<TScanColumnDesc> selectedColumns) throws Exception {
+        int offset = 0;
```
[the remark this answers is not in the document]
Comment (JNSimba) [reply samples/doris-demo/doris-source-demo/src/main/java/org/apache/doris/sdk/DorisReaderExample.java:184]: The offset here is at the level of a single tablet, and the result is the data of all tablets

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by smallhibiscus — APPROVED** [review]

**Comment by JNSimba** [comment]
run buildall

**Pull request merged**